src/coreclr/src/debug/daccess/request.cpp

```diff
@@ -4741,3 +4798,202 @@ HRESULT ClrDataAccess::GetBreakingChangeVersion(int* pVersion)
     *pVersion = SOS_BREAKING_CHANGE_VERSION;
     return S_OK;
 }
+
+HRESULT ClrDataAccess::GetObjectComWrappersData(CLRDATA_ADDRESS objAddr, CLRDATA_ADDRESS *rcw, unsigned int count, CLRDATA_ADDRESS *mowList, unsigned int *pNeeded)
+{
+#ifdef FEATURE_COMWRAPPERS
+    if (objAddr == 0 )
+    {
+        return E_INVALIDARG;
+    }
+
+    if (count > 0 && mowList == NULL)
+    {
+        return E_INVALIDARG;
+    }
+
+    SOSDacEnter();
+    auto ComWrapperCallback = [](void *mocw, void *additionalData)
+    {
+        CQuickArrayList<CLRDATA_ADDRESS> *comWrappers = (CQuickArrayList<CLRDATA_ADDRESS>*)additionalData;
+
+        comWrappers->Push(TO_CDADDR(mocw));
+
+    };
+
+    // We could bail due to missing memory, want to return S_FALSE in that case
```

Comment:
Note that SOSDacLeave assigns `hr`, so in the case of missing memory I don't think this function will return S_FALSE.

Reply:
Doesn't SOSDacLeave only reassign the hr if there is an exception that is caught? Then returning whatever the exception has for an hr should be fine

Reply:
It's totally fine to return the `hr` set by the exception, but missing memory _is_ propagated as an exception.  I'm only pointing out that your comment here "we return S_FALSE when we encounter missing memory" isn't true.  Either the comment should say "we return E_WhateverItIs when we bail due to missing memory" or remove the comment entirely...

Reply:
I think I was originally planning on wrapping it all in a EX_TRY_ALLOW_DATATARGET_MISSING_MEMORY when I wrote that. I removed the comments and changed it so it returns S_FALSE if it fails to get any data